### PR TITLE
ci(build): Get libtexpdf into build using flake inputs

### PR DESCRIPTION
**Pull request by doronbehar**
After experimenting with building projects that depend on git submodules, I reached the conclusion that this is the most comfortable way of handling such dependencies. You may like this approach since updating `libtexpdf` for the flakes will only require:

```
nix flake update
```

and the above will also update libtexpdf to it's latest version. Using a specific version of libtexpdf is possible using:

```
nix build --override-input libtexpdf-src "github:sile-typesetter/libtexpdf/my_feature_branch"
```

You can also do the above while disabling changing the `flake.lock` file:

```
nix build --override-input libtexpdf-src "github:sile-typesetter/libtexpdf/my_feature_branch" --no-write-lock-file
```

### CODE REVIEW

**Comment by alerque** [comment]
Thanks! This makes perfect sense and if Nix was the only thing in play here I'd be all for it. As it is I'm a bit unsure since this workflow now splits the workflows that update the upstream source build vs. the Nix flake. I am not sure I want it to be a separate action that could be out of sync. I'll probably fiddle with a way to get them both updated at the same time or otherwise kept in sync. Just triggering `nix flake update` at the same time seems a bit aggressive since it updates a whole bunch of other pinned things.

File: Makefile.am [hunk 1]
```diff
@@ -130,8 +127,7 @@ sile-%.md: CHANGELOG.md
 update_libtexpdf:
 	$(GIT) diff-index --quiet --cached HEAD || exit 1 # die if anything already staged
 	$(GIT) submodule update --init --remote -- libtexpdf
-	$(GIT) submodule status -- libtexpdf | $(AWK) '{ print $$1 }' > libtexpdf.git-rev
-	$(GIT) add -- libtexpdf libtexpdf.git-rev
+	$(GIT) add -- libtexpdf
```
[the remark this answers is not in the document]
Comment (doronbehar) [reply Makefile.am:130]: Oh sorry I missed your top comment of the review.

**Comment by doronbehar** [comment]
There's also the option:

```
nix flake lock --commit-lock-file --override-input libtexpdf-src github:sile-typesetter/libtexpdf/$(git submodule status -- libtexpdf | awk '{print $1}')
```

**Review by alerque — APPROVED** [review]

**Comment by alerque** [comment]
Yes I put it through a few paces by downgrading libtexpdf and updating it again, it seemed to work as expected. I had to remove the `--commit-lock-file` because it didn't play nice with our commit message requirements, but also it is nice to have it be part of the same commit as the submodule anyway. Thanks for the help with this!

**Pull request merged**